### DISCUSSION TITLE
Upgrading Airflow from v2.0.1 to v2.2.2

### DISCUSSION THREAD

**Discussion by saip50** [apache/airflow#19741] [Q&A]
I am running Apache Airflow(2.0.1) on an Azure AKS. I am trying to upgrade it to Airflow(2.2.2) using new version of chart (1.3.0), but getting an issue "CrashLoopBackoff" as pods are not airflow-migration/airflow-scheduler pods initializing. we are using our custom image pulling from our azure container registry, updated secrets in values.yaml too. I am using executor type as kubernetes executor. 
I am not able to see logs of failed pod too but getting console exception: Error: INSTALLATION FAILED: failed post-install: timed out waiting for the condition.
**DEBUG**:
install.go:178: [debug] Original chart version: ""
install.go:199: [debug] CHART PATH: C:\Users\MA652KK\AppData\Local\Temp\helm\repository\airflow-1.3.0.tgz

client.go:299: [debug] Starting delete for "airflow-broker-url" Secret
client.go:128: [debug] creating 1 resource(s)
client.go:299: [debug] Starting delete for "airflow-fernet-key" Secret
client.go:128: [debug] creating 1 resource(s)
client.go:128: [debug] creating 17 resource(s)
client.go:299: [debug] Starting delete for "airflow-run-airflow-migrations" Job
client.go:128: [debug] creating 1 resource(s)
client.go:528: [debug] Watching for changes to Job airflow-run-airflow-migrations with timeout of 5m0s
client.go:556: [debug] Add/Modify event for airflow-run-airflow-migrations: ADDED
client.go:595: [debug] airflow-run-airflow-migrations: Jobs active: 1, jobs failed: 0, jobs succeeded: 0
Error: INSTALLATION FAILED: failed post-install: timed out waiting for the condition
helm.go:88: [debug] failed post-install: timed out waiting for the condition
INSTALLATION FAILED
main.newInstallCmd.func2
        helm.sh/helm/v3/cmd/helm/install.go:127
github.com/spf13/cobra.(*Command).execute
        github.com/spf13/cobra@v1.2.1/command.go:856
github.com/spf13/cobra.(*Command).ExecuteC
        github.com/spf13/cobra@v1.2.1/command.go:974
github.com/spf13/cobra.(*Command).Execute
        github.com/spf13/cobra@v1.2.1/command.go:902
main.main
        helm.sh/helm/v3/cmd/helm/helm.go:87
runtime.main
        runtime/proc.go:225
runtime.goexit
        runtime/asm_amd64.s:1371

**Comment by potiuk** [comment]
You nneed to get and analyse the logs of the "Crash Loopback" pods and analyse it. EKS/kubectl should allow you to do that. I recommend for examlple `k9s` tool (you fill find it googling)  that helps to see the logs in an intutive way. 
Without knowing what crashed, you won't find the solution.

**Comment by saip50** [comment]
yeah, I got the logs of "Crash Loopback" pods:
**airflow-scheduler pod:** 
Error from server (BadRequest): container "scheduler" in pod "airflow-scheduler-7dc9d4d49c-zq8vl" is waiting to start: PodInitializing.

**airflow-webserver pod:** 
Error from server (BadRequest): container "webserver" in pod "airflow-webserver-7d957c64d6-nz2gp" is waiting to start: PodInitializing.

**airflow-migration pod:** 
BACKEND=postgresql
DB_HOST=localhost
DB_PORT=5432
....................
ERROR! Maximum number of retries (20) reached.
       while checking postgresql connection.

Last check result:

**Reply by potiuk** [reply]
Looks like your postgres connection is wrong in the migration job then.

**Reply by saip50** [reply]
yeah, the issue was with postgres config and now I have pods running i.e scheduler, postgres-db, migration pods are running but facing DAG issue with webserver pod. attached the screenshot of the webserver pod. Do I need add any config related to dag in values.yaml, not sure.
![MicrosoftTeams-image (1)](https://user-images.githubusercontent.com/89022002/142886518-3ca1b3e3-659e-4774-b330-cb92ebe84a8c.png)
![MicrosoftTeams-image](https://user-images.githubusercontent.com/89022002/142886528-5dffc7ae-cabd-4986-8bfb-99280aed3d8b.png)

And also, should this parameter be true/false if we are using our custom image. let me know.
![image](https://user-images.githubusercontent.com/89022002/142887461-1d19dc1d-2616-4670-8d6b-a7412d694610.png)

**Reply by potiuk** [reply]
Your migration did not run. I think you might have some race conditions or misconfiguration of your deployment - but I think you need to debug it yourself first, before involving others in debugging your deployment issues. You need to look at the logs of your migration job.

Alternatively (I do not know how you deploy airflow), you should switch to the comunity managed Helm Chart https://airflow.apache.org/docs/helm-chart/stable/index.html  and docker image, rather than relying on your own. This will give you much higher chance that someone will be able to help - without digging in your deployment problems.

**Reply by potiuk** [reply]
> To avoid images with user code, you can turn this to 'true' and all the 'run-airflow-migrations' and 'wait-for-airflow-migrations' containers/jobs will use the images from 'defaultAirflowRepository:defaultAirflowTag' values to run and wait for DB migrations .

**Reply by potiuk** [reply]
Up to you what you use. I do not know what kind of customisations you have in the image and whether they can break the migration. Again - careful look at your mirgration logs might tell you if this is your image that broke it - but you need to understand your deployment to make such decisions yourself.
